service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCMOController.java

```diff
@@ -0,0 +1,146 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.google.common.collect.ImmutableMap;
+import io.swagger.annotations.Api;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.enums.CMOActionType;
+import uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrderAction;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.service.CaseManageOrderActionService;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+
+import java.io.IOException;
+import java.util.Map;
+
+@Api
+@RestController
+@RequestMapping("/callback/action-cmo")
+public class ActionCMOController {
+    private static final String CASE_MANAGEMENT_ORDER_ACTION_KEY = "caseManagementOrderAction";
+    private static final String CASE_MANAGEMENT_ORDER_KEY = "caseManagementOrder";
+
+    private final DraftCMOService draftCMOService;
+    private final CaseManageOrderActionService caseManageOrderActionService;
+    private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+
+    public ActionCMOController(DraftCMOService draftCMOService,
+                               CaseManageOrderActionService caseManageOrderActionService,
+                               DocmosisDocumentGeneratorService docmosisDocumentGeneratorService,
+                               UploadDocumentService uploadDocumentService) {
+        this.draftCMOService = draftCMOService;
+        this.caseManageOrderActionService = caseManageOrderActionService;
+        this.docmosisDocumentGeneratorService = docmosisDocumentGeneratorService;
+        this.uploadDocumentService = uploadDocumentService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> caseDataMap = caseDetails.getData();
+
+        draftCMOService.prepareCustomDirections(caseDetails.getData());
```

Comment:
what does prepareCustomDirections do? Is there not a method from SDO that would do this?

Reply:
out of scope for 24